src/AppBundle/Command/DocsPublishCommand.php

```diff
@@ -8,47 +8,33 @@
 use Symfony\Component\Console\Input\InputOption;
 use Symfony\Component\Console\Output\OutputInterface;
 
-class DocsPublishCommand extends ContainerAwareCommand
-{
-    protected function configure()
-    {
-        $this
-          ->setName('docs:publish')
-          ->setDescription('...')
-          ->addArgument('paths', InputArgument::IS_ARRAY,
-            'One or more book expressions ("book/lang/branch"). (Default: all)');
-    }
-
-    protected function execute(InputInterface $input, OutputInterface $output)
-    {
-        /** @var \AppBundle\BookLoader $books */
-        $books = $this->getContainer()->get('book.loader');
-
-        /** @var \AppBundle\Utils\Publisher $publisher */
-        $publisher = $this->getContainer()->get('publisher');
-
-        $rows = $books->findAsList();
-
-        if (empty($rows)) {
-            $output->writeln("<error>No books found</error>");
-        }
-
-        $rowKeys = $input->getArgument('paths') ? $input->getArgument('paths') : array_keys($rows);
-
-        foreach ($rowKeys as $rowKey) {
-            $row = $rows[$rowKey];
-            $output->writeln("");
-            $output->writeln(sprintf("Publish [%s/%s/%s] (from %s)",
-              $row['book'], $row['lang'], $row['branch'], $row['repo']));
-
-            $publisher->publish($row['book'], $row['lang'], $row['branch']);
-
-            foreach ($publisher->getMessages() as $message) {
-                $output->writeln($message['label'] . ': ' . $message['content']);
-            }
-            $publisher->clearMessages();
-
-        }
+class DocsPublishCommand extends ContainerAwareCommand {
+
+  protected function configure() {
+    $this
+      ->setName('docs:publish')
+      ->setDescription('Publish one or more books')
+      ->addArgument(
+          'identifiers',
+          InputArgument::IS_ARRAY,
+          'One or more book identifiers (e.g. "user/en/master"). Partial '
+          . 'identifiers are acceptable (e.g. "user/en" will publish all '
+          . 'English versions of the User Guide. If no identifiers are '
+          . 'specified, then all versions of all languages in all books will '
+          . 'be published.');
+  }
+
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    /** @var \AppBundle\Utils\Publisher $publisher */
+    $publisher = $this->getContainer()->get('publisher');
+    $identifiers = $input->getArgument('identifiers');
```

Comment:
@totten thanks for taking a look at this :slightly_smiling_face: 

Short answer: the app infers the books/languages/versions from the yaml files for all the books. 

Long answer: 

1. First of all, there's a bug here which I just noticed (and maybe why you're commenting?) that prevents a user from publishing everything via `docs:publish`. Just [fixed here](https://github.com/civicrm/civicrm-docs/pull/45/commits/1ab4d257305105931487bd1f40261912c0d278bd) 
1. If no identifiers are specified, then [`Publisher::publish()`](https://github.com/seanmadsen/civicrm-docs/blob/refactor-model/src/AppBundle/Utils/Publisher.php#L407) is called with no arguments. You can probably see what happens from that point on.